### PR TITLE
Refactor vIOMMU test code

### PR DESCRIPTION
**Pull request by Yingshun**
Test results - aarch64- build 20: all passed in jenkins or on local machine
image
 (1/1) type_specific.io-github-autotest-libvirt.vIOMMU.iommu_device_settings.virtio_muti_devices.vhost_on.smmuv3: PASS (84.02 s)
 (1/1) type_specific.io-github-autotest-libvirt.vIOMMU.iommu_device_lifecycle.reboot_many_times.hostdev_iface.virtio: PASS (340.88 s)

Test results - x86_64- build 17: all passed in jenkins or on local machine except the last one(ip cmd timeout in guest, it's not related to this pr)
image
 (1/1) type_specific.io-github-autotest-libvirt.sriov.plug_unplug.attach_detach_interface_special_situations.with_model: PASS (73.83 s)
 (1/1) type_specific.io-github-autotest-libvirt.sriov.vm_lifecycle.reboot.hostdev_interface.vf_address.failover.with_iommu: PASS (288.50 s)
 (1/2) type_specific.io-github-autotest-libvirt.sriov.failover.at_dt_hostdev.hostdev_interface: PASS (139.77 s)

### CODE REVIEW

File: provider/viommu/viommu_base.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        return disk_dict
+
+    def setup_iommu_test(self, **dargs):
+        """
+        iommu test environment setup
+
+        :param dargs: Other test keywords
+        """
+        iommu_dict = dargs.get('iommu_dict', {})
+        test_scenario = dargs.get('test_scenario', '')
```
Comment (cliping) [inline provider/viommu/viommu_base.py:133]: Unused parameter?

File: provider/viommu/viommu_base.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            self.test.log.info("TEST_SETUP: Add iommu device.")
+            libvirt_virtio.add_iommu_dev(self.vm, iommu_dict)
+
+    def teardown_iommu_test(self, **dargs):
+        """
+        Cleanup iommu test environment
+
+        :param dargs: Other test keywords
+        """
+        test_scenario = dargs.get('test_scenario', '')
```
Comment (cliping) [inline provider/viommu/viommu_base.py:150]: Same as above.

**Review by cliping — COMMENTED** [review]
Others LGTM

File: provider/viommu/viommu_base.py (hunk 1 above)
Comment (Yingshun) [reply provider/viommu/viommu_base.py]: yes, updated.

**Review by cliping — APPROVED** [review]

**Review by chloerh — APPROVED** [review]

**Pull request merged**